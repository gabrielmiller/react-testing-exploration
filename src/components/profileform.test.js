import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import ProfileForm from './profileform';
import TestButton from './testButton';

test('it is a contrived assertion', () => {
    expect(true).toBe(true);
});

test('contains a button when shallow rendered', () => {
    const wrapper = shallow(<ProfileForm />);
    expect(wrapper.find(TestButton)).toHaveLength(1);
});

test('contains two text inputs', () => {
    const wrapper = shallow(<ProfileForm />);
    expect(wrapper.find('input[type="text"]')).toHaveLength(2);
});

test('it matches the snapshot', () => {
    const component = renderer.create(<ProfileForm />).toJSON();
    expect(component).toMatchSnapshot();
});