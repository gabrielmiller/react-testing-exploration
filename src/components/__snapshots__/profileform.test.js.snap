// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="first-name"
    >
      First Name
    </label>
    <input
      id="first-name"
      name="first-name"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="last-name"
    >
      Last Name
    </label>
    <input
      id="last-name"
      name="last-name"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div>
    <button
      className="button-lg"
      type="submit"
    >
      Create User
    </button>
  </div>
</form>
`;
